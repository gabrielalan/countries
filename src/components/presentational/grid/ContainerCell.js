import React from 'react';

export default function ContainerCell({ columnSpec, width, data }) {
  return <div className="grid__row__column" style={{ width }}>
    <columnSpec.Type>{data}</columnSpec.Type>
  </div>
}
