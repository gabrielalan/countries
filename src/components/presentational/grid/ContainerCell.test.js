import React from 'react';
import renderer from 'react-test-renderer';
import ContainerCell from './ContainerCell';

const TypeOne = ({ children }) => `${children} One`;
const TypeTwo = ({ children }) => `${children} Two`;

describe('ContainerCell', () => {
  it('should render correct type', () => {
    const treeOne = renderer.create(
      <ContainerCell
        data="Type ==="
        columnSpec={{ Type: TypeOne }} />
    ).toJSON();

    const treeTwo = renderer.create(
      <ContainerCell
        data="Type ==="
        columnSpec={{ Type: TypeTwo }} />
    ).toJSON();

    expect(treeOne).toMatchSnapshot();
    expect(treeTwo).toMatchSnapshot();
  });

  it('should set the width', () => {
    const tree = renderer.create(
      <ContainerCell
        data="check width"
        width={100}
        columnSpec={{ Type: TypeOne }} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
