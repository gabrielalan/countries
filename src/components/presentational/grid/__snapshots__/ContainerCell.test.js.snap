// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerCell should render correct type 1`] = `
<div
  className="column"
  style={
    Object {
      "width": undefined,
    }
  }
>
  Type === One
</div>
`;

exports[`ContainerCell should render correct type 2`] = `
<div
  className="column"
  style={
    Object {
      "width": undefined,
    }
  }
>
  Type === Two
</div>
`;

exports[`ContainerCell should set the width 1`] = `
<div
  className="column"
  style={
    Object {
      "width": 100,
    }
  }
>
  check width One
</div>
`;
